documentation/api/api-versioning.md

```diff
@@ -0,0 +1,56 @@
+changes that would break our clients from calling our API (eg. making a field required that was previously optional).
+
+By having versioned endpoints, we can give our clients time to update their calls until the old version of the endpoint can be fully deprecated.
+
+# Current Implementation
+
+Endpoints are versioned in their routes. For example, if our `GET` opportunity endpoint is `GET /v1/opportunities/:opportunityId` and
+we made a change that was not backwards compatible (let's say we change the authentication approach), then we'd make an entirely
+new endpoint called `GET /v2/opportunities/:opportunityId`. The `v1` endpoint would still exist, and most of the logic would likely be shared
+between the implementations, but they would be two entirely separate endpoints as far as API routing is concerned.
+
+While we are in the early phases of developing the API, prior to having any significant users of the API, we won't worry about
+maintaining the backwards compatability. We won't start adding new versions of the API until we would begin impacting production
+systems of our users.
+
+# Long-term idea using headers
+
+An alternative approach we could take in the future is to instead put the versioning in a header parameter (like the `content-type` field) similar to [Stripe](https://stripe.com/blog/api-versioning).
+
+However, that approach has a few complexities due to the libraries we use for defining schemas and endpoints:
+
+* [webargs](https://webargs.readthedocs.io/en/latest/index.html) - a generic framework for parsing HTTP requests, and can serialize/deserialize using Marshmallow schemas.
+* [apispec](https://apispec.readthedocs.io/en/latest/) - a tool that can generate OpenAPI specifications from Marshmallow schemas.
+* [APIFlask](https://apiflask.com/) - the framework we use that connects Flask, webargs, and apispec together for us.
+
+When we construct a route, we specify the input & output Marshmallow schemas, which APIFlask passes to webargs & apispec. The OpenAPI specification is generated entirely at app start-up, and
+is a static file (we also generate the [file statically](https://github.com/HHS/simpler-grants-gov/blob/main/api/openapi.generated.yml) in our repo using the same underlying code).
+For webargs, it registers the schemas with the routes, and when the internal routing of a request is going on, it will use the registered schema to do validation.
+
+Webargs does seem to have support for choosing the schema for an endpoint during request handling, as the "schema" can be either a schema or a `callable` object that
+takes in the request and returns a `Schema` object. That request object contains the header, so a callable could be used to choose the right schema for a users given request.
+
+However, the `input` and `output` methods for APIFlask do not take in multiple Schema or a callable, and would need to be rewritten.
```

Comment:
Interesting to know that there's a potential workaround, but the juice is probably not worth the squeeze until we can find a significant use case for providing more fine-grained control over specifying the API version for a given API call that wouldn't be satisfied by path-based versioning.